tutorials/using-cloud-vpn-with-cisco-asr/index.md

```diff
@@ -0,0 +1,893 @@
+use security services to communicate securely. During tunnel establishment, the
+two peers negotiate security associations that govern authentication,
+encryption, encapsulation, and key management. These negotiations involve two
+phases: first, to establish the tunnel (the IKE SA) and second, to govern
+traffic within the tunnel (the IPsec SA). The following commands set the SA
+lifetime and timing parameters.
+
+* `! IPsec SA lifetime` – 1 hour is the recommended value on ASR 1000 router.
+
+* `! IPsec SA replay window-size` – 1024 is the recommended value on ASR 1000 router.
+
+    crypto ipsec security-association lifetime seconds 3600
+
+    crypto ipsec security-association replay window-size 1024
+
+#### Configure IPsec transform set
+
+A transform set represents a certain combination of security protocols and
+algorithms. During the IPsec SA negotiation, the peers agree to use a particular
+transform set for protecting a particular data flow.
+
+    crypto ipsec transform-set VPN_SCALE_TEST_TS esp-aes 256 esp-sha-hmac mode tunnel
+
+#### Configure IPsec profile
+
+Defines the IPsec parameters that are to be used for IPsec encryption between
+two IPsec routers in IPsec profile configuration. In this block, the following
+parameters are set
+
+* Perfect Forward Secrecy (PFS) - PFS ensures that the same key will not be
+  generated again, so forces a new diffie-hellman key exchange. Set to group16
+  as recommended configuration on ASR 1000 router.
+* SA Lifetime - set the lifetime of the security associations (after which a
+  reconnection will occur). Set to `3600 seconds` as recommended configuration
+  on ASR 1000 router.
+
+    crypto ipsec profile VPN_SCALE_TEST_VTI
+     set security-association lifetime seconds 3600
+     set transform-set VPN_SCALE_TEST_TS
+     set pfs group16
+     set ikev2-profile VPN_SCALE_TEST_IKEV2_PROFILE
+
+#### Configure IPsec static virtual tunnel interface (SVTI)
+
+A tunnel interface is configured to be the logical interface associated with the
+tunnel. All traffic routed to the tunnel interface will be encrypted and
+transmitted to the GCP. Similarly, traffic from the GCP will be logically
+received on this interface.
+
+Association with the IPsec security association is done through the
+`tunnel protection` command.
+
+Adjust the maximum segment size (MSS) value of TCP packets going through a
+router. The recommended value is 1360 when the number of IP MTU bytes is set to
+1400. With these recommended settings, TCP sessions quickly scale back to
+1400-byte IP packets so the packets will "fit" in the tunnel.
+
+    !
+    interface Tunnel1
+     ip address 169.254.0.58 255.255.255.252
+     ip mtu 1400
+     ip tcp adjust-mss 1360
+     tunnel source TenGigabitEthernet0/0/0
+     tunnel mode ipsec ipv4
+     tunnel destination 104.196.200.68
+     tunnel protection ipsec profile VPN_SCALE_TEST_VTI
+    !
+
+#### Configure static or dynamic routing protocol to route traffic into the IPsec tunnel
+
```

Comment:
Just wanted to double if this configuration follows the best practices: https://cloud.google.com/router/docs/resources/best-practices